components/org.wso2.carbon.identity.application.authentication.handler.session/src/main/java/org/wso2/carbon/identity/application/authentication/handler/session/ActiveSessionsLimitHandler.java

```diff
@@ -494,18 +494,20 @@ private String getUserTenantDomain(AuthenticationContext context) throws Authent
         return tenantDomain;
     }
 
-    private boolean isSessionActiveInSameBrowser(HttpServletRequest request, List<UserSession> userSessions) {
-
-        Cookie cookie = FrameworkUtils.getAuthCookie(request);
-        if (cookie == null) {
-            return false;
-        }
-        String sessionContextKey = DigestUtils.sha256Hex(cookie.getValue());
-        boolean isSessionMatch = userSessions.stream()
-                .anyMatch(userSession -> sessionContextKey.equals(userSession.getSessionId()));
-        if (isSessionMatch && log.isDebugEnabled()) {
-            log.debug("Session context key matches with the session ID: " + sessionContextKey);
+    /**
+     * Check whether the current authentication attempt is a single sign-on attempt.
+     *
+     * @param context      Authentication context.
+     * @param userSessions List of user sessions.
+     * @return True if the current authentication attempt is a single sign-on attempt.
+     */
+    private boolean isSingleSignOnAttempt(AuthenticationContext context, List<UserSession> userSessions) {
+
+        String sessionIdFromContext = context.getSessionIdentifier();
+        if (userSessions != null && StringUtils.isNotBlank(sessionIdFromContext)) {
```

Comment:
`userSessions != null` we don't need to check this since it is already checked in Line 221

Reply:
Fixed with https://github.com/wso2-extensions/identity-local-auth-basicauth/pull/227/commits/635524b8c374d865913a49bdcc18a2800a283fcf